packages/DataTable/src/DataTable.tsx

```diff
@@ -149,59 +155,47 @@ const DataTable: React.FC<DataTableProps> & DataTableComposition = ({
     );
   }
 
-  function renderTable(height: number, width: number | string) {
-    return (
+  if (extractedResizeContainer) {
+    console.warn("The `DataTable.ResizeContainer` has been deprecated, please use `height` and `width` prop instead.");
+  }
+
+  return (
+    <ThemeContext.Provider value={{ borderType, isHeaderSticky, hasZebraStripes }}>
       <sc.Table
         ref={tableRef}
         aria-label={a11yText}
         aria-rowcount={data.length}
         data-pka-anchor="dataTable"
-        width={width}
-        height={height}
+        width={bestDimensions.width}
+        height={bestDimensions.height}
         {...tableInstance.getTableProps()}
         {...moreProps}
       >
         <sc.ContentWrapper>
           <RenderRowContext.Provider value={renderRow}>
-            <ReactTableContext.Provider value={tableInstance}>{renderTableContent(height)}</ReactTableContext.Provider>
+            <ReactTableContext.Provider value={tableInstance}>
+              {renderTableContent(bestDimensions.height)}
+            </ReactTableContext.Provider>
           </RenderRowContext.Provider>
         </sc.ContentWrapper>
       </sc.Table>
-    );
-  }
-
-  if (!extractedResizeContainer) {
-    if (width === undefined) {
-      console.warn("The `width` prop should be specified if `<DataTable.ResizeContainer>` is not used");
-    }
-
-    if (height === undefined) {
-      console.warn("The `height` prop should be specified if `<DataTable.ResizeContainer>` is not used");
-    }
-  }
-
-  return (
-    <ThemeContext.Provider value={{ borderType, isHeaderSticky, hasZebraStripes }}>
-      {extractedResizeContainer ? (
-        <div {...extractedResizeContainer.props}>
-          <ResizeDetector isFullHeight>
-            {({ height: maxHeight, width: maxWidth }) =>
-              renderTable(getTableHeight(maxHeight), getTableWidth(maxWidth))
-            }
-          </ResizeDetector>
-        </div>
-      ) : (
-        renderTable(height || 0, width || 0)
-      )}
     </ThemeContext.Provider>
   );
-};
+}
+
+declare module "react" {
+  function forwardRef<T, P = unknown>(
+    render: (props: P, ref: React.RefObject<T>) => React.ReactElement | null
+  ): React.FC<DataTableProps> & DataTableComposition;
+}
+
+const DataTableWithRef = React.forwardRef(DataTable);
 
-DataTable.displayName = "DataTable";
-DataTable.InfiniteLoader = InfiniteLoader;
-DataTable.ResizeContainer = ResizeContainer;
-DataTable.types = {
+DataTableWithRef.displayName = "DataTable";
+DataTableWithRef.InfiniteLoader = InfiniteLoader;
+DataTableWithRef.ResizeContainer = ResizeContainer;
```

Comment:
Should I just remove this export `DataTableWithRef.ResizeContainer` ? cause the prop changes will be breaking changes anyway, but if I removed it we can't really extract the children component and show warnings

Reply:
Yes, let's just remove it, and add a migration instructions to the changelog